server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -973,6 +1071,9 @@ private void indexValue(DocumentParserContext context, long timestamp) {
             DataStreamTimestampFieldMapper.storeTimestampValueForReuse(context.doc(), timestamp);
         }
 
+        if (hasDocValuesSkipper && hasDocValues) {
```

Comment:
Here the doc values skipper has precedence over the `index` parameter only when the setting is actually enabled.

Reply:
I was looking where `setDocValuesType(...)` gets invoked, but I see that `SortedNumericDocValuesField.indexedField(...)` adding doc value with doc value skipper enabled.